libr/core/rvc.c

```diff
@@ -0,0 +1,115 @@
+#include <rvc.h>
+static bool copy_commits(const Rvc *repo, const char *dpath, const char *sname) {
+	char *path, *name, *spath;
+	spath = r_str_newf ("%s" R_SYS_DIR "branches" R_SYS_DIR "%s" R_SYS_DIR "commits", repo->path, sname);
+	r_return_val_if_fail (spath, false);
+	RList *files = r_sys_dir (spath);
+	if (!files) {
+		free (spath);
+		return false;
+	}
+	RListIter *iter;
+	ls_foreach (files, iter, name) {
+		path = r_str_newf ("%s%s", spath, sname);
+		if (!path) {
+			free (spath);
+			r_list_free (files);
+			return false;
+		}
+		if (!r_file_copy (dpath, path)) {
+			free (spath);
```

Comment:
Just break

Reply:
If I break the function will return true no matter what and thats a bad idea, if you really want to minimize code a goto here would be a good idea.

Reply:
I added a ret variable